src/index.js

```diff
@@ -392,6 +392,54 @@ async function getEntries({ input, cwd }) {
 	return entries;
 }
 
+function replaceName(filename, name) {
+	return resolve(
+		dirname(filename),
+		name + basename(filename).replace(/^[^.]+/, ''),
+	);
+}
+
+function getMain({ options, entry, format }) {
+	const { pkg } = options;
+	const pkgMain = options['pkg-main'];
+
+	if (!pkgMain) {
```

Comment:
I wonder if we could even apply this same logic if the value provided for the `--output` argument has a file extension. Like `microbundle -f umd -o bundle.js` would see `.js` and know to use that name as-is.

Reply:
I think this would open to questions:
- Such a new logic could be a "breaking" change because there might be people already using -o with ".js" and expecting it to add the format suffix. If a breaking change would mean that this has to wait for 1.0 I personally would prefer to split this idea into another issue 😇
- If we implement this logic, it should only ever apply if the format option is set with a single value

Reply:
Agree with @katywings here. I'm worried that us making too many assumptions will lead to unexpected results for the user. In either case this would be a breaking change.